gnovm/pkg/gnopkgfetch/gnopkgfetch_testing.go

```diff
@@ -0,0 +1,99 @@
+package gnopkgfetch
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/gnolang/gno/tm2/pkg/amino"
+	tm2client "github.com/gnolang/gno/tm2/pkg/bft/rpc/client"
+	ctypes "github.com/gnolang/gno/tm2/pkg/bft/rpc/core/types"
+	types "github.com/gnolang/gno/tm2/pkg/bft/rpc/lib/types"
+	"github.com/gnolang/gno/tm2/pkg/sdk"
+)
+
+func InjectExamplesClient(t *testing.T) {
+	t.Helper()
+
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		t.Fatalf("failed to get source path")
+	}
+	examplesDir := filepath.Join(filepath.Dir(filename), "..", "..", "..", "examples")
+	oldClient := fetchClient
+	fetchClient = tm2client.NewRPCClient(&examplesMockClient{examplesRoot: examplesDir})
+	t.Cleanup(func() {
+		fetchClient = oldClient
+	})
+}
+
+type examplesMockClient struct {
+	examplesRoot string
+}
+
+func (m *examplesMockClient) SendRequest(ctx context.Context, request types.RPCRequest) (*types.RPCResponse, error) {
```

Comment:
Can we clean up this mock a bit? 🙏

Reply:
It's very hard to read

Reply:
I'll clean up if you validate the new way it's injected :) otherwise I'll probably have to scrap it

Reply:
refactored in 8d0a5a28da5a51dab22e7e8dfbea1f5ffbcb4a38